src/test/java/io/supertokens/test/AuthRecipeTest.java

```diff
@@ -205,8 +205,11 @@ public void paginationTest() throws Exception {
         }
 
         UserInfo user1 = EmailPassword.signUp(process.getProcess(), "test0@example.com", "password0");
+        Thread.sleep(1); // Sleep to remove race condition
```

Comment:
One of the times that I ran this test, the test failed because when the users were sorted by creation time, they were not sorted in the correct order. This is due to Java's implementation of `System.currentTimeMillis()` that does not necessarily return a unique timestamp on each call.

See this excerpt from the Java documentation:
> `public static long currentTimeMillis()`
> 
> Returns the current time in milliseconds. Note that while the unit of time of the return value is a millisecond, the granularity of the value depends on the underlying operating system and may be larger. For example, many operating systems measure time in units of tens of milliseconds.

For Example:
```java
for(int i = 0; i < 50; i++)
    System.out.println(System.currentTimeMillis());
```
When I ran this code on my Windows machine, it printed out `1650942035022` many times, followed by `1650942035026` many more times.

However, if I run this second example:
```java
for(int i = 0; i < 50; i++) {
    System.out.println(System.currentTimeMillis());
    Thread.sleep(1);
}
```
The above prints out 50 unique timestamps, 1-3 milliseconds apart. This is due to Java's implementation of `Thread.sleep()`, which is guaranteed to sleep *at least* the number of milliseconds that it is given, dependent on the resolution of the clock of the underlying operating system.